test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2101,9 +2101,24 @@ protected static SecureRandom secureRandomFips(final byte[] seed) throws NoSuchA
         return secureRandomFips;
     }
 
+    /**
+     * The timeout used for the various "safe" wait methods such as {@link #safeAwait} and {@link #safeAcquire}. In tests we generally want
+     * these things to complete almost immediately, but sometimes the CI runner executes things rather slowly so we use {@code 10s} as a
+     * fairly relaxed definition of "immediately".
+     * <p>
+     * A well-designed test should not need to wait for longer than this. If you think you need to do so, instead seek a better way to write
+     * the test such that it does not need to wait for so long. Tests that take 10s of seconds to complete are a big drag on CI times which
```

Comment:
>the test such that it does not need to wait for so long

May be we should give a hint(s) on those?
For example when waiting for an es timeout to happen we should generally ensure the test configures a custom fairly short timeout rather than relying on generous elasticsearch default one.

Reply:
IMO tests which wait for a timeout should be using a fake clock so that we can set a realistic timeout duration without having to wait for that long in real time. I know we have quite a lot of tests that set, and wait for, a timeout in the region of ~100ms, but this is not really testing a realistic situation and might therefore miss some bugs, whilst still being an approach that does not scale well when we're talking about a test suite the size of ES's.